### PR TITLE
137 send to address change address

### PR DESCRIPTION
**Pull request by dwalintukan**
**Update qtum 0.14.16 when this is merged in**

### CODE REVIEW

File: src/api/blockchain.js [hunk 1]
```diff
@@ -27,6 +27,18 @@ const Blockchain = {
     return qClient.getBlockCount();
   },
 
+  async getBlockHash(args) {
```
Comment (alex-cory) [inline src/api/blockchain.js]: you might already know this, but you can destructure this in the props.
i.e.
```
async getBlockHash({ blockNum }) {
  ...
}
```

Comment (dwalintukan) [reply src/api/blockchain.js]: true, just wanted to follow the same structure all the other functions for now. will do a refactor later to change all of them at once.

File: src/schema/index.js [hunk 2]
```diff
@@ -59,6 +59,7 @@ type Transaction {
   type: _TransactionType!
   status: _TransactionStatus!
   txid: String
+  createdBlock: Int
```
Comment (frankobe) [inline src/schema/index.js]: what's the difference between `createdBlock` & `blockNum`?

File: package.json [hunk 3]
```diff
@@ -122,7 +122,7 @@
     "nedb-promise": "^2.0.1",
     "node-fetch": "^1.7.3",
     "pg": "^7.4.0",
-    "qweb3": "^0.7.5",
+    "qweb3": "file:../qweb3.js/qweb3-0.7.6.tgz",
```
[the remark this answers is not in the document]
Comment (dwalintukan) [reply package.json]: done

File: src/route/api.js [hunk 4]
```diff
@@ -118,6 +118,15 @@ apiRouter.get('/get-block-count', (req, res, next) => {
     });
 });
 
+apiRouter.post('/get-block-hash', (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (dwalintukan) [reply src/route/api.js]: this is just for my dev purposes, so i can do a quick req from postman. gonna keep it as get-block-hash to match naming scheme. block and hash are separate words

File: src/schema/index.js (hunk 2 above)
Comment (dwalintukan) [reply src/schema/index.js]: `createdBlock` is for the block when the tx is created. we only track `createdTime` right now but i needed this for `transfer` qtum tx. 

they changed something in qtum client where it doesnt return the block num anymore in `gettransaction` which is how i validate transfer qtum txs. need to know the createdBlock to be able to compare it for confirmation purposes.

Comment (frankobe) [reply src/schema/index.js]: @dwalintukan i c. Need find a new rpc call then. Can we add a `TODO` on this line and I track it #199

File: src/sync/update_tx.js [hunk 5]
```diff
@@ -40,13 +40,15 @@ async function updatePendingTxs(db, currentBlockCount) {
...
     const txInfo = await wallet.getTransaction({ txid: tx.txid });
-    const blockInfo = await blockchain.getBlock({ blockHash: txInfo.blockhash });
 
-    if (currentBlockCount >= blockInfo.height + Config.TRANSFER_MIN_CONFIRMATIONS) {
-      tx.status = txInfo.confirmations === 0 ? txState.FAIL : txState.SUCCESS;
+    if (txInfo.confirmations > 0) {
+      tx.status = txState.SUCCESS;
       tx.gasUsed = Math.floor(Math.abs(txInfo.fee) / Config.DEFAULT_GAS_PRICE);
+
+      const blockHash = await blockchain.getBlockHash({ blockNum: currentBlockCount });
```
Comment (frankobe) [inline src/sync/update_tx.js]: I think it should be 
```
tx.blockNum = currentBlockCount - txInfo. confirmations;
const blockHash = await blockchain.getBlockHash({ blockNum: tx.blockNum });
const blockInfo = await blockchain.getBlock({ blockHash });
tx.blockTime = blockInfo.time;
```

Comment (dwalintukan) [reply src/sync/update_tx.js]: done

File: src/sync/update_tx.js [hunk 6]
```diff
@@ -47,9 +47,9 @@ async function updateTx(tx, currentBlockCount) {
       tx.status = txState.SUCCESS;
       tx.gasUsed = Math.floor(Math.abs(txInfo.fee) / Config.DEFAULT_GAS_PRICE);
 
+      tx.blockNum = currentBlockCount - txInfo.confirmations + 1;
       const blockHash = await blockchain.getBlockHash({ blockNum: currentBlockCount });
```
Comment (frankobe) [inline src/sync/update_tx.js]: `const blockHash = await blockchain.getBlockHash({ blockNum: tx.blockNum });`

File: src/sync/update_tx.js [hunk 7]
```diff
@@ -40,14 +40,16 @@ async function updatePendingTxs(db, currentBlockCount) {
...
 
-    if (currentBlockCount >= blockInfo.height + Config.TRANSFER_MIN_CONFIRMATIONS) {
-      tx.status = txInfo.confirmations === 0 ? txState.FAIL : txState.SUCCESS;
+    if (txInfo.confirmations > 0) {
+      tx.status = txState.SUCCESS;
       tx.gasUsed = Math.floor(Math.abs(txInfo.fee) / Config.DEFAULT_GAS_PRICE);
-      tx.blockNum = blockInfo.height;
+
+      tx.blockNum = currentBlockCount - txInfo.confirmations + 1;
+      const blockHash = await blockchain.getBlockHash({ blockNum: currentBlockCount });
```
Comment (frankobe) [inline src/sync/update_tx.js]: `const blockHash = await blockchain.getBlockHash({ blockNum: tx.blockNum });`

Comment (dwalintukan) [reply src/sync/update_tx.js]: done

Comment (frankobe) [reply src/sync/update_tx.js]: @dwalintukan changed the wrong line

Comment (dwalintukan) [reply src/sync/update_tx.js]: fixed

**Pull request merged**